generators/app/index.js

```diff
@@ -155,20 +244,44 @@ module.exports = yeoman.Base.extend({
 
     var donePrompting = this.async();
     this.prompt(prompts, function (props) {
-      this.sdk = this.sdkVersions[props.sdkVersion];
-      this._saveProps([
-        constants.PROP_SDK_VERSION,
-        constants.PROP_ARCHETYPE_VERSION,
-        constants.PROP_PROJECT_GROUP_ID,
-        constants.PROP_PROJECT_ARTIFACT_ID,
-        constants.PROP_PROJECT_VERSION,
-        constants.PROP_PROJECT_PACKAGE,
-        constants.PROP_COMMUNITY_OR_ENTERPRISE,
-        constants.PROP_REMOVE_DEFAULT_SOURCE_AMPS,
-        constants.PROP_REMOVE_DEFAULT_SOURCE_SAMPLES,
-      ], props);
-      // can only setup module registry/manager once we have other variables setup
-      this.moduleManager = require('./alfresco-module-manager.js')(this);
+      // The below 2 if conditions are here because the prompts used in testing
+      // are not injected into the prompts
+      if(null != props[constants.PROP_WARNING] && props[constants.PROP_WARNING]) {
+        this.bail = true;
+      }
+      if(null != props[constants.PROP_CREATE_SUB_FOLDER] && !props[constants.PROP_CREATE_SUB_FOLDER]) {
+        this.bail = true;
+      }
+      if(!this.bail) {
+        props.generatorVersion = this.pkg.version;
+        if (!this.config.get(constants.PROP_ORIGINAL_GENERATOR_VERSION)) {
+          props[constants.PROP_ORIGINAL_GENERATOR_VERSION] = this.pkg.version;
+        }
+        if(null != props.projectPath && process.cwd() != props.projectPath) {
+          if (!fs.existsSync(props.projectPath)) {
+            fs.mkdirSync(props.projectPath);
+          }
+          process.chdir(props.projectPath);
+          this.destinationRoot(props.projectPath);
+        }
+        this.sdk = this.sdkVersions[props.sdkVersion];
+        this._saveProps([
+          constants.PROP_ORIGINAL_GENERATOR_VERSION,
+          constants.PROP_GENERATOR_VERSION,
+          constants.PROP_SDK_VERSION,
+          constants.PROP_ARCHETYPE_VERSION,
+          constants.PROP_PROJECT_GROUP_ID,
+          constants.PROP_PROJECT_ARTIFACT_ID,
+          constants.PROP_PROJECT_PATH,
```

Comment:
Actually, by searching for this constant, it appears that it is used to set this property but the property is never read, so maybe we can get rid of it altogether? I'd like that :)